src/Features/CSharp/Portable/UsePatternMatching/CSharpAsAndNullCheckDiagnosticAnalyzer.cs

```diff
@@ -56,94 +68,44 @@ private void SyntaxNodeAction(SyntaxNodeAnalysisContext syntaxContext)
                 return;
             }
 
-            var severity = styleOption.Notification.Value;
-
-            // look for the form "if (a != null)" or "if (null != a)"
-            var ifStatement = (IfStatementSyntax)syntaxContext.Node;
-
-            // "x is Type y" is only available in C# 7.0 and above.  Don't offer this refactoring
-            // in projects targeting a lesser version.
-            if (((CSharpParseOptions)ifStatement.SyntaxTree.Options).LanguageVersion < LanguageVersion.CSharp7)
-            {
-                return;
-            }
-
-            // If has to be in a block so we can at least look for a preceding local variable declaration.
-            if (!ifStatement.Parent.IsKind(SyntaxKind.Block))
-            {
-                return;
-            }
-
-            // We need to find the leftmost expression in the if-condition.  If this is a
-            // "x != null" expression, then we can replace it with "o is Type x".  
-            var condition = GetLeftmostCondition(ifStatement.Condition);
-            if (!condition.IsKind(SyntaxKind.NotEqualsExpression))
-            {
-                return;
-            }
-
-            // look for the form "x != null" or "null != x".
-            if (!IsNullCheckExpression(condition.Left, condition.Right) &&
-                !IsNullCheckExpression(condition.Right, condition.Left))
-            {
-                return;
-            }
-
-            var conditionName = condition.Left is IdentifierNameSyntax
-                ? (IdentifierNameSyntax)condition.Left
-                : (IdentifierNameSyntax)condition.Right;
-
-            // Now make sure the previous statement is "var a = ..."
-            var parentBlock = (BlockSyntax)ifStatement.Parent;
-            var ifIndex = parentBlock.Statements.IndexOf(ifStatement);
-            if (ifIndex == 0)
-            {
-                return;
-            }
-
-            var previousStatement = parentBlock.Statements[ifIndex - 1];
-            if (!previousStatement.IsKind(SyntaxKind.LocalDeclarationStatement))
+            var ifOrWhileStatement = (StatementSyntax)node;
+            
+            // if/while has to be in a block so we can at least look for a preceding local variable declaration.
+            if (!(ifOrWhileStatement.Parent is BlockSyntax parentBlock))
```

Comment:
I wish there was a way to get the corresponding SyntaxKind from the syntax node type